### PR TITLE
[marked] Add InlineLexer Type

### PR DESCRIPTION
**Pull request by htkzhtm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/markedjs/marked/blob/master/lib/marked.js#L640
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/marked/index.d.ts [hunk 1]
```diff
@@ -77,6 +77,16 @@ declare namespace marked {
      */
     function setOptions(options: MarkedOptions): typeof marked;
 
+    class InlineLexer {
+        constructor(links: string[], options?: MarkedOptions);
+        rules: Rules;
```
Comment (BendingBender) [inline types/marked/index.d.ts]: `rules` is also exposed as a static property.

File: types/marked/index.d.ts [hunk 2]
```diff
@@ -77,6 +77,16 @@ declare namespace marked {
...
+        output(src: string): string;
```
Comment (BendingBender) [inline types/marked/index.d.ts]: `output` is also exposed as a static method, albeit with a different signature.

File: types/marked/index.d.ts [hunk 3]
```diff
@@ -77,6 +77,16 @@ declare namespace marked {
...
+        escapes(text: string): string;
```
Comment (BendingBender) [inline types/marked/index.d.ts]: `escapes` is only exposed as a static method, it's not an instance method.

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Comment by htkzhtm** [comment]
Thank you for reviewing. Static property added, and fix test.

**Review by BendingBender — APPROVED** [review]

**Pull request merged**